src/reconcile/rec_visibility.c

```diff
@@ -382,15 +410,25 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
              * keep the same on-disk value but set the stop time pair to indicate that the validity
              * window ends when this tombstone started.
              */
-            WT_ERR(__rec_append_orig_value(session, page, tombstone, vpack));
```

Comment:
Why you revert this change?

Reply:
I added a condition in `__rec_append_orig_value` function to not to add on-disk prepared value back to the update list, because there exists a prepared update already on the list. But I need to add it when it is a prepared TOMBSTONE that is written to disk. Using the same function in this code flow is leading it problem to prepare. I find the old code is simple to handle it.

Reply:
I'm not sure why it is hard to add the condition to `__rec_append_orig_value ` but I feel it will be harder to maintain if we have prepared update cases handled in different places. Also there are failures regarding this part of the code (WT-6093). We should first deal with that otherwise if we revert this, we may miss some problems.

Reply:
I thought of changing the `__rec_append_orig_value` function, but then I have to remove the new asserts that are added as part of WT-6073 because the on-disk `vpack` and `upd` may have different timestamps when it is a prepared update after it resolved. If you don't interest with that assert I can remove and use that function.

Reply:
Can you fix the assert instead? I think we can still assert that if it is a prepared update.

Reply:
No, It can't be fixed, because the in-memory update loses the prepare timestamp when that prepare update once it is resolved and on-disk version has only the prepare timestamp, so it can't match whether it is the same update for which the on-disk version is written. We can retain the assert and validate it only for non-prepared updates?

Reply:
Yes, this is what I mean actually. We just verify non-prepared updates.

Reply:
Why in-memory update loses the prepare timestamp when that prepare update once it is resolved and why we only have the prepare timestamp on the disk image for in-memory prepared updates?

Reply:
The in-memory update stores the prepare timestamp in `start_ts` once the update is resolved, it is replaced with a commit timestamp. But the written on-disk version contains the information of prepared timestamp as part of the cell.